test/routing/query-params.test.js

```diff
@@ -1,72 +1,128 @@
 var test  = require('tape')
   , rhumb = require('../../src/rhumb')
+  , utils = require('../utils')
 
 test("Routing should pass query string params to callback", function(t) {
-  var router = rhumb.create()
+  t.plan(2)
 
-  t.plan(1)
-  router.add("/sing", function(params) {
-    t.deepEquals(params,
-      { foo: "bar"
-      , baz: "bam"
-      }
-    )
+  utils.performMatch('/foo', '/foo?q=value', function (params) {
+    t.deepEquals(params, { q: 'value' }
+      , 'path should match with query params included in params list')
   })
 
-  router.match("/sing?foo=bar&baz=bam")
+  utils.performMatch('/foo/bar/bing', '/foo/bar/bing?q=value', function (params) {
+    t.deepEquals(params, { q: 'value' }
+      , 'path should match with query params included in params list')
+  })
 })
 
 test("Routing should pass query string params and path params to callback", function(t) {
-  var router = rhumb.create()
-
   t.plan(1)
-  router.add("/sing/{sound}", function(params) {
-    t.deepEquals(params,
-      { sound : "bird-song"
-      , foo : "bar"
-      , baz : "bam"
-      }
-    )
-  })
 
-  router.match("/sing/bird-song?foo=bar&baz=bam")
+  utils.performMatch('/wibble/{wobble}', '/wibble/foo?q=value', function (params) {
+    t.deepEqual(params, { wobble: 'foo', q: 'value' }
+      , 'path should match without overwriting path params with query params')
+  })
 })
 
 test("Routing should add query string params without values as undefined", function(t) {
-  var router = rhumb.create()
-
   t.plan(1)
-  router.add("/sing/{sound}", function(params) {
-    t.deepEquals(params,
-      { sound : "bird-song"
-      , foo : undefined
-      , bar : undefined
-      , baz : "bop"
-      }
-    )
-  })
 
-  router.match("/sing/bird-song?foo=&bar&baz=bop")
+  utils.performMatch('/wibble', '/wibble?foo=&bar&bing=value', function (params) {
+    t.deepEqual(params, { foo: undefined, bar: undefined, bing: 'value' }
+      , 'path should match with empty and non-empty query params included in params list')
+  })
 })
 
 test("Routing should handle incomplete query strings", function(t) {
-  var router = rhumb.create()
+  t.plan(1)
+
+  utils.performMatch('/wibble', '/wibble?foo=bar&bing&', function (params) {
+    t.deepEqual(params, { foo: 'bar', bing: undefined }
+      , 'path should match with complete and incomplete query params included in params list')
```

Comment:
Again here, `bing` should probably be `''` or `null` (we need to just decide on which is the empty value.)